File: py/240.py
class Solution:
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if len(matrix)==0 or len(matrix[0])==0:
            return False;
        row=len(matrix)-1;
        col=0;
        while row>=0 and col<len(matrix[0]):
            if matrix[row][col]==target:
                return True;
            elif matrix[row][col]<target:
                col+=1;
            elif matrix[row][col]>target:
                row-=1;
        return False;

File: C++/240.cpp
#include "mod.h"
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        if(matrix.empty()||matrix[0].empty())return false;
        int row=matrix.size()-1, col=0;
        while(row>=0&&col<matrix[0].size()){
            if(matrix[row][col]==target)return true;
            else if(matrix[row][col]>target)--row;
            else if(matrix[row][col]<target)++col;
        }
        return false;
    }
};